src/NuGet.Protocol.Core.v3/DataClient/DataClient.cs

```diff
@@ -124,6 +124,11 @@ public JObject GetJObject(Uri address)
         public async Task<JObject> GetJObjectAsync(Uri address, CancellationToken token)
         {
             var response = await GetAsync(address, token);
+            if (response.StatusCode == System.Net.HttpStatusCode.Unauthorized)
```

Comment:
Other resources are calling GetJObjectAsync and would not be expecting this. Either update the other resources to handle this, or use GetAsync instead and check the status code directly in the index.json resource.